optimade/models/optimade_json.py

```diff
@@ -23,6 +26,102 @@
 )
 
 
+class DataType(Enum):
+    """Optimade Data Types
+
+    See the section "Data types" in the OPTIMADE API specification for more information.
+    """
+
+    STRING = "string"
+    INTEGER = "integer"
+    FLOAT = "float"
+    BOOLEAN = "boolean"
+    TIMESTAMP = "timestamp"
+    LIST = "list"
+    DICTIONARY = "dictionary"
+    UNKNOWN = "unknown"
+
+    @classmethod
+    def get_values(cls):
+        """Get OPTIMADE data types (enum values) as a (sorted) list"""
+        return sorted((_.value for _ in cls))
+
+    @classmethod
+    def from_python_type(cls, python_type: Union[type, str, object]):
+        """Get OPTIMADE data type from a Python type"""
+        mapping = {
```

Comment:
Could we define these mappings (incl. JSON map below) as properties?
e.g.

```
@property
def python_to_optimade_type_map(...):
```

and 
```
@property
json_to_optimade_type_map(...)
```

Reply:
Tried to do this, the issue is that then it's not reachable on the class level, i.e., `DataType.from_json_type("number")` doesn't work anymore.
If they're both instance methods/properties, then you first have to instantiate it, i.e., `DataType("float").from_json_type("date-time")`, which is quite confusing, since `"float"` is completely ignored in this process.

Do you have a smart way of doing this? Other than just making it a static (class) method perhaps?

Reply:
Yeah, I'm trying to do it with static methods now, but that of course doesn't work since we're utilizing the self referenced enumerations ...
Could I make the mappings enumerations? Or should they simply be outside the Enum?

Reply:
Ah, hmm, didn't appreciate this when I left my suggestion. In that case, I would leave it as is for now; I like the classmethod access and this is a clean way of doing it.

I'll profile the manipulation of our models at some point so we can target our optimisations, then we can decide where to put the effort in.